src/lib.rs

```diff
@@ -2429,6 +2429,116 @@ pub fn undirected_gnm_random_graph(
     Ok(graph)
 }
 
+#[inline]
+fn pnorm(x: f64, p: f64) -> f64 {
+    if p == 1.0 || p == std::f64::INFINITY {
+        x.abs()
+    } else if p == 2.0 {
+        x * x
+    } else {
+        x.abs().powf(p)
+    }
+}
+
+fn distance(x: &[f64], y: &[f64], p: f64) -> f64 {
+    let it = x.iter().zip(y.iter()).map(|(xi, yi)| pnorm(xi - yi, p));
+
+    if p == std::f64::INFINITY {
+        it.fold(-1.0, |max, x| if x > max { x } else { max })
+    } else {
+        it.sum()
+    }
+}
+
+/// Returns a random geometric graph in the unit cube of dimensions `dim`.
+///
+/// The random geometric graph model places `num_nodes` nodes uniformly at
+/// random in the unit cube. Two nodes are joined by an edge if the
+/// distance between the nodes is at most `radius`.
+///
+/// Each node has a node attribute ``'pos'`` that stores the
+/// position of that node in Euclidean space as provided by the
+/// ``pos`` keyword argument or, if ``pos`` was not provided, as
+/// generated by this function.
+///
+/// :param int num_nodes: The number of nodes to create in the graph
+/// :param float radius: Distance threshold value
+/// :param int dim: Dimension of node positions. Default: 2
+/// :param list pos: Optional list with node positions as values
+/// :param float p: Which Minkowski distance metric to use.  `p` has to meet the condition
+///     ``1 <= p <= infinity``.
+///     If this argument is not specified, the :math:`L^2` metric
+///     (the Euclidean distance metric), p = 2 is used.
+/// :param int seed: An optional seed to use for the random number generator
+///
+/// :return: A PyGraph object
+/// :rtype: PyGraph
+#[pyfunction]
+#[text_signature = "(num_nodes, radius, dim=2, pos=None, p=2.0, seed=None, /)"]
+pub fn random_geometric_graph(
+    py: Python,
+    num_nodes: usize,
+    radius: f64,
+    dim: Option<usize>,
```

Comment:
We can make this input type just `usize` and handle the default for python in the macro definition

```suggestion
    dim: usize,
```